lib/src/strings.dart

```diff
@@ -72,6 +72,15 @@ String _yamlEncodeDoubleQuoted(String string) {
 /// It is important that we ensure that [string] is free of unprintable
 /// characters by calling [assertValidScalar] before invoking this function.
 String _tryYamlEncodeSingleQuoted(String string) {
+  // If [string] contains a newline we'll use double quoted strings instead.
+  // Single quoted strings can represent newlines, but then we have to use an
+  // empty line (replace \n with \n\n). But since leading spaces following
+  // line breaks are ignored, we can't represent "\n ".
+  // Thus, if the string contains `\n` and we're asked to do single quoted,
+  // we'll fallback to a double quoted string.
+  if (string.contains('\n')) {
```

Comment:
Perhaps better to update `_hasUnprintableCharacters` to also include newlines?

Reply:
That'll make all folded strings that contain newlines into double quoted strings...

I think this part of the code needs some refactoring. There are probably more bugs in this area.